authentification/database/database.go

```diff
@@ -0,0 +1,38 @@
+package database
+
+import (
+	"context"
+	"log"
+	"os"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+)
+
+func GetMongoDbCollection(DbName string, CollectionName string) (*mongo.Collection, error) {
+	client, err := GetMongoDbConnection()
+
+	if err != nil {
+		return nil, err
+	}
+
+	collection := client.Database(DbName).Collection(CollectionName)
+
+	return collection, nil
+}
+
+func GetMongoDbConnection() (*mongo.Client, error) {
+	client, err := mongo.Connect(context.Background(), options.Client().ApplyURI(os.Getenv("MONGO_URI")))
+
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Essaie de pas trop mettre de fatal dans des layers qui peuvent être profonds, gère l'erreur avec verbosité pour donner un contexte quand ton service fail.
Si y'a un 2e db mongoDB dans ton service, on sait pas forcément laquelle à fail.

```suggestion
		return nil,fmt.Errorf("Database mongoDB for (user ?) failed to connect: %v", err)
```